nuxeo-features/nuxeo-elasticsearch/nuxeo-elasticsearch-core/src/main/java/org/nuxeo/elasticsearch/listener/ElasticSearchInlineListener.java

```diff
@@ -106,10 +106,12 @@ public void beforeCompletion() {
 
     @Override
     public void afterCompletion(int status) {
-        if (getAllCommands().isEmpty()) {
-            return;
-        }
+        // un hook the current listener even if there's no commands to index
+        // unless, during next transaction this listener won't be hooked to it
```

Comment:
not sure to understand the "unless" part and why it is related to the indexing of root document